openfl/utilities/mocks.py

```diff
@@ -0,0 +1,12 @@
+# Copyright (C) 2020-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Mock objects to eliminate extraneous dependencies"""
+
+
+class MockDataLoader:
```

Comment:
I suppose you don't extend `DataLoader`, because only the `get_feature_shape()` method is required during plan initialization? In this case, would it make sense to call this class `FeatureShapeLoader`?